packages/nx/src/command-line/yargs-utils/shared-options.ts

```diff
@@ -407,3 +408,15 @@ const coerceTuiAutoExit = (value: string) => {
   }
   throw new Error(`Invalid value for --tui-auto-exit: ${value}`);
 };
+
+function concurrency(str: string) {
+  let parallel = parseInt(str);
```

Comment:
`args['parallel']` and `args['maxParallel']` can come in as a number. We need to check if it's a number of string first.